File: python/interview_preparation_kit/problem_1_repeated_string.py
import time


def repeatedString(string, length):
    a_count_in_string = string.count("a")
    num_full_repeats, remaining_length = divmod(length, len(string))
    total_repeats = num_full_repeats * a_count_in_string + string[
        :remaining_length
    ].count("a")
    return total_repeats


if __name__ == "__main__":
    string = "a"
    length = 1000000000000
    start_time = time.time()
    result = repeatedString(string, length)
    print(result)
    elapsed_time = time.time() - start_time
    print(f"--- {elapsed_time:.4f} seconds ---")

File: Rust/src/interview_preparation_kit/problem_1_repeated_string.rs
pub fn repeat_string(s: &str, n: i64) -> i64 {
  let a_count_in_string = s.chars().filter(|&c| c == 'a').count() as i64;
  let (num_full_repeat, remaining_length) = (n / s.len() as i64, n % s.len() as i64);
  let total_repeats = num_full_repeat * a_count_in_string
      + s.chars()
          .take(remaining_length as usize)
          .filter(|&c| c == 'a')
          .count() as i64;
  total_repeats
}